File: Interop/fic/oneArray_fic.c
#include <stdio.h>

/* C function to call a routine in fortran, */
/*   it defines an array with N elements and call */
/*   a routine to sum it's elements */

extern void vecref_(int * , int[], int *);

int main(void)
  
{
  int sum,i, N = 12;
  int v[N];

  for (i = 0; i < N; ++i) {
    v[i] = 1;
  }

  vecref_(&N,v, &sum);

  printf("%d \n", sum);
  
  return 0;
}

File: Interop/fic/oneArray_fic.f90
subroutine VecRef(N,v,total)
  ! Subroutine to serve as a module for a C function,
  !   it gets the number of elements N of a vector v and
  !   returns the total sum of its elements

  integer :: N
  integer ::i, total
  integer :: v(N)

  total = 0

  do i = 1,N
     total = total + v(i)
  end do

end subroutine VecRef


